### ISSUE TITLE
Upgrade to Jetty 9.3.0

### ISSUE THREAD

**Issue by arteam** [dropwizard/dropwizard#1036]
This is an issue for future versions (maybe 1.0), because currently we are not ready for migration.

The Jetty development team actively works on 9.3.0 version. The main feature is HTTP/2 support, but also there are also many performance improvements and bugfixes .

I've been playing around with the last milestone build (9.3.0.M2) on HTTP/2 support in Dropwizard. The current work in this [branch](https://github.com/arteam/dropwizard/tree/jetty-93). Basically, I managed to create a module and start Dropwizard as an HTTP/2 server. But during this process I found several places, where we are not compatible:

* `org.eclipse.jetty.servlet.ServletTester` was removed
* `dropwizard-metrics-jetty9` doesn't support 9.3.0
* `SPDY` is not sipported in Jetty 9.3.0
* `GzipFilter` was removed (actually deprecated, but the code is stripped) and migrated to `GzipHandler`
* In `RequestLog` the signature of the `log` method is changed.

I plan to update the list according to the latest Dropwizard and Jetty milestones and convert this
issue to a pull request when we are ready.

**Comment by jplock** [comment]
Relates to https://github.com/dropwizard/metrics/pull/823

**Comment by joakime** [comment]
`org.eclipse.jetty.servlet.ServletTester` was not removed.

It's available still, but as part of the `<classifier>tests</classifier>` artifact.

```
<dependency>
    <groupId>org.eclipse.jetty</groupId>
    <artifactId>jetty-servlet</artifactId>
    <version>9.3.0.v20150612</version>
    <classifier>tests</classifier>
</dependency>
```

**Comment by arteam** [comment]
Yes, thanks. I also figured it out. The branch I mentioned is a bit of outdated, because I didn't push the latest changes, but locally I added *tests* dependencies on `jetty-servlet` and `jetty-http` to use `ServletTester` and `HttpTester`.

**Comment by devmage** [comment]
@arteam I pulled down your fork and merged master with your branch. Only conflict was with the `dropwizard-spdy/pom.xml` that you removed, which was trivially resolved (by removing it again). All current tests passed.

**Comment by arteam** [comment]
Thanks. I've rebased the changes against current master and it should be mergeable now.

I didn't have time to work on this and 9.3.* support is not ready yet. Most of the items in the list are done, except HTTP/2, so probably some applications could be run on Jetty 9.3.* with this branch.

I plan to make to more rigor testing. If I everything is stable enough, I will send a pull request it to the upstream as soon as we release 0.9. The HTTP/2 support will probably become a separate issue, because of it's complexity.

**Comment by arteam** [comment]
Closed by #1330
